src/swupd/files_test.go

```diff
@@ -0,0 +1,178 @@
+package swupd
+
+import "testing"
+
+func TestSetFileTypeFromFlagFile(t *testing.T) {
+	flags := map[string]string{
+		"F": "file",
+		"D": "directory",
+		"L": "link",
+		"M": "manifest",
+		".": "",
+	}
+
+	for flag, ftype := range flags {
```

Comment:
This is not a typical style for a test. See [wiki](https://github.com/golang/go/wiki/TableDrivenTests) or [cheney's blog](https://dave.cheney.net/2013/06/09/writing-table-driven-tests-in-go) for a more normal style.